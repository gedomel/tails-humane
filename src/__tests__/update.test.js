import React from 'react';
import renderer from 'react-test-renderer';
import UpdateForm from '../Components/update_form';
import Update from '../Components/update_animal';

/**
 * Test Method to ensure that the update form renders as expected.
 */
it('Render An Animal Update Correctly', () => {
  const tree = renderer.create(<Update />).toJSON();
  expect(tree).toMatchSnapshot();
});

/**
 * Test Method to ensure updating an Availablity to Unavailable on the form renders correctly.
 */
it('Render An Unavailable Dog Status Update Correctly', () => {
  const tree = renderer
    .create(
      <UpdateForm
        name="Mango"
        animalID="395468"
        availablility="Unavailable"
        housetrained="Yes"
        noDogs="OK"
        noCats="OK"
        noKids="OK"
        species="Dog"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/**
 * Test Method to ensure updating an Availablity to Available on the form renders correctly.
 */
it('Render An Available Cat Status Update Correctly', () => {
  const tree = renderer
    .create(
      <UpdateForm
        name="Mr. Man"
        animalID="7888890"
        availablility="Available"
        housetrained="Yes"
        noDogs="OK"
        noCats="OK"
        noKids="No kids"
        species="Cat"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
