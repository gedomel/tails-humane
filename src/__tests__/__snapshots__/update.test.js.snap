// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render An Animal Update Correctly 1`] = `
<div>
  <div
    className="form"
  >
    <h2>
       Update an Animal
    </h2>
    <p>
      Enter Animal ID:
    </p>
    <form
      className="form"
      onSubmit={[Function]}
    >
      <input
        className="post_put"
        onChange={[Function]}
        placeholder="i.e 8909809"
        type="text"
      />
      <button
        className="SubmitButton"
        type="submit"
      >
        Find
      </button>
    </form>
  </div>
  <div />
</div>
`;

exports[`Render An Available Cat Status Update Correctly 1`] = `
<div
  className="updateform"
>
  <form
    onSubmit={[Function]}
  >
    <p>
      Name:
      Mr. Man
    </p>
    <p>
      TAILS ID:
      7888890
    </p>
    <p>
      Species:
      Cat
    </p>
    <p>
      Select Availability:
    </p>
    <div>
      <select
        onChange={[Function]}
        value="Available"
      >
        <option
          value="Available"
        >
          Available
        </option>
        <option
          value="Unavailable"
        >
          Unavailable
        </option>
      </select>
      <p>
        Housetrained?:
      </p>
      <select
        onChange={[Function]}
        value="Yes"
      >
        <option
          value="Partially"
        >
          Partially
        </option>
        <option
          value="Yes"
        >
          Yes
        </option>
        <option
          value="No"
        >
          No
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Dogs?:
      </p>
      <select
        onChange={[Function]}
        value="OK"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No dogs"
        >
          No dogs
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Cats?:
      </p>
      <select
        onChange={[Function]}
        value="OK"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No cats"
        >
          No cats
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Kids?:
      </p>
      <select
        onChange={[Function]}
        value="No kids"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No small kids"
        >
          No small kids
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
    </div>
    <button
      className="SubmitButton"
      type="submit"
    >
      Submit Changes
    </button>
    <p
      className="success"
    >
      
    </p>
  </form>
</div>
`;

exports[`Render An Unavailable Dog Status Update Correctly 1`] = `
<div
  className="updateform"
>
  <form
    onSubmit={[Function]}
  >
    <p>
      Name:
      Mango
    </p>
    <p>
      TAILS ID:
      395468
    </p>
    <p>
      Species:
      Dog
    </p>
    <p>
      Select Availability:
    </p>
    <div>
      <select
        onChange={[Function]}
        value="Unavailable"
      >
        <option
          value="Available"
        >
          Available
        </option>
        <option
          value="Unavailable"
        >
          Unavailable
        </option>
      </select>
      <p>
        Housetrained?:
      </p>
      <select
        onChange={[Function]}
        value="Yes"
      >
        <option
          value="Partially"
        >
          Partially
        </option>
        <option
          value="Yes"
        >
          Yes
        </option>
        <option
          value="No"
        >
          No
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Dogs?:
      </p>
      <select
        onChange={[Function]}
        value="OK"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No dogs"
        >
          No dogs
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Cats?:
      </p>
      <select
        onChange={[Function]}
        value="OK"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No cats"
        >
          No cats
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
      <p>
        No Kids?:
      </p>
      <select
        onChange={[Function]}
        value="OK"
      >
        <option
          value="OK"
        >
          OK
        </option>
        <option
          value="No small kids"
        >
          No small kids
        </option>
        <option
          value="Unknown"
        >
          Unknown
        </option>
      </select>
    </div>
    <button
      className="SubmitButton"
      type="submit"
    >
      Submit Changes
    </button>
    <p
      className="success"
    >
      
    </p>
  </form>
</div>
`;
